test/number_test.cpp

```diff
@@ -63,35 +63,43 @@ TEST(NumberTest, ScaleFloatingPoint)
 TEST(NumberTest, MapIntegral)
 {
   EXPECT_EQ(ksn::map(5, 2, 4, 0, 1), 1) <<
-    "5 - 2 = 3\n"
+    "min(5, max(2, 4)) = 4\n"
+    "max(4, min(2, 4)) = 4\n"
+    "4 - 2 = 2\n"
     "4 - 2 = 2\n"
     "1 - 0 = 1\n"
-    "scale(3, 2, 1) = 1\n"
+    "scale(2, 2, 1) = 1\n"
     "1 + 0 = 1";
 
-  EXPECT_EQ(ksn::map(0, -2, -4, 0, 1), -1) <<
-    "0 - (-2) = 2\n"
+  EXPECT_EQ(ksn::map(0, -2, -4, 0, 1), 0) <<
+    "min(0, max(-2, -4)) = -2\n"
+    "max(-2, min(-2, -4)) = -2\n"
+    "-2 - (-2) = 0\n"
     "-4 - (-2) = -2\n"
     "1 - 0 = 1\n"
-    "scale(2, -2, 1) = -1\n"
-    "-1 + 0 = -1";
+    "scale(0, -2, 1) = 0\n"
+    "0 + 0 = 0";
```

Comment:
FYI, the lines after `<<` are only for debugging, you can simply remove them and focus on adding more assertions:
```c++
EXPECT_EQ(ksn::map(5, 2, 4, 0, 1), 1);
EXPECT_EQ(ksn::map(0, -2, -4, 0, 1), 0);
```